frontend/src/components/header/MenuOverlay.js

```diff
@@ -96,18 +97,23 @@ class MenuOverlay extends Component {
     if (targetValue) {
       this.setState({
         query: e.target.value,
+        pendingQuery: e.target.value ? true : false,
       });
 
       queryPathsRequest(e.target.value, 9)
         .then((response) => {
-          this.setState({
-            queriedResults: flattenLastElem(response.data),
-          });
+          if (e.target.value === this.queryString) {
```

Comment:
extract `e.target.value` to some `query` const... just to have a better naming for it.
also IMHO u shall compare it with the query from the state and not with that weirdly call and introduced field (this.queryString), see below...